diff --git a/visa/iga/IGALibrary/CMakeLists.txt b/visa/iga/IGALibrary/CMakeLists.txt
index 53f1fe0b9..b7b133e65 100644
--- a/visa/iga/IGALibrary/CMakeLists.txt
+++ b/visa/iga/IGALibrary/CMakeLists.txt
@@ -306,7 +306,7 @@ target_compile_definitions(IGA_ENC_LIB PUBLIC IGA_DISABLE_ENCODER_EXCEPTIONS)
 
 set (IGA_RES "")
 
-    add_library(IGA_DLL SHARED $<TARGET_OBJECTS:GEDLibrary> $<TARGET_OBJECTS:IGA_OLIB> ${IGA_API} ${IGA_RES})
+    add_library(IGA_DLL STATIC $<TARGET_OBJECTS:GEDLibrary> $<TARGET_OBJECTS:IGA_OLIB> ${IGA_API} ${IGA_RES})
 
 set(IGA_DLL_DEFINITIONS ${IGA_DEFINITIONS} IGA_BUILDING_DLL)
 if (WIN32)
diff --git a/visa/iga/IGALibrary/api/kv.h b/visa/iga/IGALibrary/api/kv.h
index 9533e109e..6918d5b47 100644
--- a/visa/iga/IGALibrary/api/kv.h
+++ b/visa/iga/IGALibrary/api/kv.h
@@ -121,7 +121,7 @@ IGA_API int32_t kv_get_inst_size(const struct kv_t *kv, int32_t pc);
  *   KV_INVALID_PC if instruction cannot be found
  *   KV_DECODE_ERROR if an instruction cannot be docoded
  */
-IGA_API kv_status_t kv_get_inst_msg_info(const kv_t* kv, int32_t pc,
+IGA_API kv_status_t kv_get_inst_msg_info(const struct kv_t* kv, int32_t pc,
     bool* isAtomic, bool* isSlm, bool* isScratch);
 
 /*
@@ -289,7 +289,7 @@ IGA_API kv_status_t kv_get_send_exbso(const struct kv_t *kv, int32_t pc,
  *    KV_INVALID_ARGUMENT      if given a null parameter
  *    KV_INCAPABLE_PLATFORM    if called on unuspported platform
  */
-IGA_API kv_status_t kv_get_sendg_desc(const kv_t *kv, int32_t pc,
+IGA_API kv_status_t kv_get_sendg_desc(const struct kv_t *kv, int32_t pc,
                                       uint64_t *desc);
 
 /*
@@ -299,7 +299,7 @@ IGA_API kv_status_t kv_get_sendg_desc(const kv_t *kv, int32_t pc,
  *
  * RETURNS: similar to kv_get_sendg_desc
  */
-IGA_API kv_status_t kv_get_sendg_ind_desc0(const kv_t *kv, int32_t pc,
+IGA_API kv_status_t kv_get_sendg_ind_desc0(const struct kv_t *kv, int32_t pc,
                                            uint32_t *subregPresent,
                                            uint32_t *subreg);
 
@@ -310,7 +310,7 @@ IGA_API kv_status_t kv_get_sendg_ind_desc0(const kv_t *kv, int32_t pc,
  *
  * RETURNS: similar to kv_get_sendg_desc
  */
-IGA_API kv_status_t kv_get_sendg_ind_desc1(const kv_t *kv, int32_t pc,
+IGA_API kv_status_t kv_get_sendg_ind_desc1(const struct kv_t *kv, int32_t pc,
                                            uint32_t *subregPresent,
                                            uint32_t *subreg);
 
@@ -421,7 +421,7 @@ IGA_API uint32_t kv_get_execution_size(const struct kv_t *kv, int32_t pc);
 /*
  * Returns Software scoreboarding information.
  */
-IGA_API bool kv_get_swsb_info(const kv_t *kv, int32_t pc,
+IGA_API bool kv_get_swsb_info(const struct kv_t *kv, int32_t pc,
                               iga::SWSB_ENCODE_MODE encode_mode,
                               iga::SWSB &swsb);
 #endif
