diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5b04555b2..fb7540639 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -14,6 +14,7 @@ add_subdirectory(IGC)
 
 list(APPEND IGC__IGC_TARGETS "igc_dll")
 list(APPEND IGC__IGC_TARGETS "IGA_DLL")
+list(APPEND IGC__IGC_TARGETS "IGA_STATIC_DLL")
 list(APPEND IGC__IGC_TARGETS "IGA_EXE")
 list(APPEND IGC__IGC_TARGETS "fcl_dll")
 list(APPEND IGC__IGC_TARGETS "opencl-clang-lib")
diff --git a/visa/iga/IGALibrary/CMakeLists.txt b/visa/iga/IGALibrary/CMakeLists.txt
index 4812b330a..076119dc9 100644
--- a/visa/iga/IGALibrary/CMakeLists.txt
+++ b/visa/iga/IGALibrary/CMakeLists.txt
@@ -320,6 +320,7 @@ if(MSVC)
   )
 endif()
 
+    add_library(IGA_STATIC_DLL STATIC $<TARGET_OBJECTS:GEDLibrary> $<TARGET_OBJECTS:IGA_OLIB> ${IGA_API} ${IGC_BUILD__RES__IGC__iga_dll})
     add_library(IGA_DLL SHARED $<TARGET_OBJECTS:GEDLibrary> $<TARGET_OBJECTS:IGA_OLIB> ${IGA_API} ${IGC_BUILD__RES__IGC__iga_dll})
 
 set(IGA_DLL_DEFINITIONS ${IGA_DEFINITIONS} IGA_BUILDING_DLL)
@@ -333,6 +334,9 @@ set(IGA_DLL_LINK_FLAGS ${IGA_LINK_FLAGS})
     set_property(TARGET IGA_DLL APPEND PROPERTY
         OUTPUT_NAME                                 "iga${TARGET_MODIFIER}"
         )
+    set_property(TARGET IGA_STATIC_DLL APPEND PROPERTY
+        OUTPUT_NAME                                 "iga${TARGET_MODIFIER}"
+        )
 set_property(TARGET IGA_DLL APPEND PROPERTY
     COMPILE_DEFINITIONS                         ${IGA_DLL_DEFINITIONS}
   )
@@ -342,6 +346,15 @@ set_property(TARGET IGA_DLL APPEND PROPERTY
 set_property(TARGET IGA_DLL APPEND PROPERTY
     FOLDER                                      "IGAProjs"
   )
+set_property(TARGET IGA_STATIC_DLL APPEND PROPERTY
+    COMPILE_DEFINITIONS                         ${IGA_STATIC_DLL_DEFINITIONS}
+  )
+set_property(TARGET IGA_STATIC_DLL APPEND PROPERTY
+    LINK_FLAGS                                  "${IGA_STATIC_DLL_LINK_FLAGS}"
+  )
+set_property(TARGET IGA_STATIC_DLL APPEND PROPERTY
+    FOLDER                                      "IGAProjs"
+  )
 
 if(IGC_BUILD AND MSVC)
     # set up standard defines from the common WDK path.
@@ -408,7 +421,7 @@ endif(ANDROID AND MEDIA_IGA)
       else()
           set(IGA_LIB_PATH ${CMAKE_INSTALL_PREFIX}/lib)
       endif()
-      INSTALL(TARGETS IGA_DLL LIBRARY DESTINATION ${IGA_LIB_PATH} COMPONENT igc-core)
+      INSTALL(TARGETS IGA_DLL IGA_STATIC_DLL LIBRARY DESTINATION ${IGA_LIB_PATH} COMPONENT igc-core)
       unset(IGA_LIB_PATH)
   elseif(MSVC)
       install(TARGETS IGA_DLL RUNTIME
diff --git a/visa/iga/IGALibrary/api/iga.cpp b/visa/iga/IGALibrary/api/iga.cpp
index 9171a87c9..3607316d2 100644
--- a/visa/iga/IGALibrary/api/iga.cpp
+++ b/visa/iga/IGALibrary/api/iga.cpp
@@ -356,7 +356,7 @@ public:
     eopts.autoDepSet =
         (aopts.encoder_opts & IGA_ENCODER_OPT_AUTO_DEPENDENCIES) != 0;
     eopts.sbidCount = aopts.sbid_count;
-    eopts.swsbEncodeMode = aopts.swsb_encode_mode;
+    eopts.swsbEncodeMode = static_cast<iga::SWSB_ENCODE_MODE>(aopts.swsb_encode_mode);
 
     if ((aopts.encoder_opts & IGA_ENCODER_OPT_USE_NATIVE) == 0) {
       if (!iga::ged::IsEncodeSupported(m_model, eopts)) {
diff --git a/visa/iga/IGALibrary/api/iga.h b/visa/iga/IGALibrary/api/iga.h
index 192e046b2..201398c0b 100644
--- a/visa/iga/IGALibrary/api/iga.h
+++ b/visa/iga/IGALibrary/api/iga.h
@@ -12,8 +12,6 @@ SPDX-License-Identifier: MIT
 #ifndef IGA_H
 #define IGA_H
 
-#include "iga_types_swsb.hpp"
-
 #include <stddef.h>
 #include <stdint.h>
 #ifdef _WIN32
@@ -257,7 +255,7 @@ typedef struct {
    * Force the swsb_encode_mode.  If not given (SWSBInvalidMode),
    * then encode mode will be derived from platform.
    */
-  iga::SWSB_ENCODE_MODE swsb_encode_mode;
+  uint32_t /* iga::SWSB_ENCODE_MODE */ swsb_encode_mode;
 } iga_assemble_options_t;
 
 /* detects screwups where someone adds a field and the compiler pads
@@ -320,7 +318,7 @@ static_assert(sizeof(iga_assemble_options_t) == 8 * 4,
         0,                                          /* reserved */             \
         0,                                          /* reserved */             \
         16,                                         /* sbid_count */           \
-        iga::SWSB_ENCODE_MODE::SWSBInvalidMode                                 \
+        0                                           /* iga::SWSB_ENCODE_MODE::SWSBInvalidMode */ \
   }
 
 /*
