diff --git a/CMakeLists.txt b/CMakeLists.txt
index a11f884feb..3826e97234 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -76,6 +76,13 @@ endif()
 
 include(GNUInstallDirs)
 
+# Find sdt.h from SystemTap, install USDT probes if present
+include(CheckIncludeFile)
+check_include_file(sys/sdt.h HAVE_SDT)
+if(HAVE_SDT)
+  add_definitions(-DHAVE_SDT)
+endif()
+
 if(USE_SANITIZE_UB AND WIN32)
   add_definitions(-D_ITERATOR_DEBUG_LEVEL=0)
   set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Xlinker /NODEFAULTLIB:clang_rt.ubsan_standalone_cxx-x86_64.lib")
diff --git a/level_zero/core/source/cmdlist/cmdlist_hw_skl_to_tgllp.inl b/level_zero/core/source/cmdlist/cmdlist_hw_skl_to_tgllp.inl
index ea503ff7b3..a44baf3e34 100644
--- a/level_zero/core/source/cmdlist/cmdlist_hw_skl_to_tgllp.inl
+++ b/level_zero/core/source/cmdlist/cmdlist_hw_skl_to_tgllp.inl
@@ -14,6 +14,7 @@
 #include "shared/source/helpers/simd_helper.h"
 #include "shared/source/memory_manager/graphics_allocation.h"
 #include "shared/source/memory_manager/memory_manager.h"
+#include "shared/source/memory_manager/gfx_partition.h"
 #include "shared/source/memory_manager/residency_container.h"
 #include "shared/source/program/kernel_info.h"
 #include "shared/source/unified_memory/unified_memory.h"
@@ -24,6 +25,8 @@
 
 #include "encode_surface_state_args.h"
 
+#include "shared/source/os_interface/linux/probes.h"
+
 #include <algorithm>
 
 namespace L0 {
@@ -109,6 +112,36 @@ ze_result_t CommandListCoreFamily<gfxCoreFamily>::appendLaunchKernelWithParams(K
 
     appendEventForProfiling(event, nullptr, true, false, false, false);
 
+    if (LEVEL_ZERO_LAUNCH_KERNEL_ENABLED()) {
+        /* Get the name of the kernel */
+        size_t kernelNameSize = 0;
+        kernel->getKernelName(&kernelNameSize, NULL);
+        auto kernelName = std::make_unique<char[]>(sizeof(char) * kernelNameSize);
+        kernel->getKernelName(&kernelNameSize, kernelName.get());
+
+        /* Get the size of the kernel */
+        size_t kernelSize;
+        kernel->getKernelProgramBinary(&kernelSize, NULL);
+
+        /* Get the start pointer of the kernel */
+        auto isaAllocation = kernel->getIsaAllocation();
+        uint64_t kernelStartPointer = kernel->getIsaOffsetInParentAllocation();
+//         if (this->heaplessModeEnabled) {
+            kernelStartPointer += isaAllocation->getGpuAddress();
+//         } else {
+//             kernelStartPointer += isaAllocation->getGpuAddressToPatch();
+//         }
+
+//         if (!kernel->requiresGenerationOfLocalIdsByRuntime()) {
+//             kernelStartPointer += kernel->getKernelDescriptor().entryPoints.skipPerThreadDataLoad;
+//         }
+
+        NEO::Device *neoDevice = device->getNEODevice();
+        uint64_t iba = neoDevice->getMemoryManager()->getGfxPartition(neoDevice->getRootDeviceIndex())->getHeapBase(NEO::HeapIndex::heapInternalDeviceMemory);
+
+        LEVEL_ZERO_LAUNCH_KERNEL(iba, kernelStartPointer, kernelSize, kernelName);
+    }
+
     auto perThreadScratchSize = std::max<std::uint32_t>(this->getCommandListPerThreadScratchSize(0u),
                                                         kernel->getImmutableData()->getDescriptor().kernelAttributes.perThreadScratchSize[0]);
     this->setCommandListPerThreadScratchSize(0u, perThreadScratchSize);
diff --git a/level_zero/core/source/cmdlist/cmdlist_hw_xehp_and_later.inl b/level_zero/core/source/cmdlist/cmdlist_hw_xehp_and_later.inl
index 710be73210..fff70ce06e 100644
--- a/level_zero/core/source/cmdlist/cmdlist_hw_xehp_and_later.inl
+++ b/level_zero/core/source/cmdlist/cmdlist_hw_xehp_and_later.inl
@@ -15,6 +15,7 @@
 #include "shared/source/indirect_heap/indirect_heap.h"
 #include "shared/source/kernel/grf_config.h"
 #include "shared/source/memory_manager/memory_manager.h"
+#include "shared/source/memory_manager/gfx_partition.h"
 #include "shared/source/memory_manager/residency_container.h"
 #include "shared/source/memory_manager/unified_memory_manager.h"
 #include "shared/source/program/kernel_info.h"
@@ -33,6 +34,8 @@
 #include "encode_surface_state_args.h"
 #include "igfxfmid.h"
 
+#include "shared/source/os_interface/linux/probes.h"
+
 namespace L0 {
 
 template <GFXCORE_FAMILY gfxCoreFamily>
@@ -317,6 +320,35 @@ ze_result_t CommandListCoreFamily<gfxCoreFamily>::appendLaunchKernelWithParams(K
         appendEventForProfilingAllWalkers(compactEvent, nullptr, launchParams.outListCommands, true, true, launchParams.omitAddingEventResidency, false);
     }
 
+    if (LEVEL_ZERO_LAUNCH_KERNEL_ENABLED()) {
+        /* Get the name of the kernel */
+        size_t kernelNameSize = 0;
+        kernel->getKernelName(&kernelNameSize, NULL);
+        auto kernelName = std::make_unique<char[]>(sizeof(char) * kernelNameSize);
+        kernel->getKernelName(&kernelNameSize, kernelName.get());
+
+        /* Get the size of the kernel */
+        size_t kernelSize;
+        kernel->getKernelProgramBinary(&kernelSize, NULL);
+
+        /* Get the start pointer of the kernel */
+        auto isaAllocation = kernel->getIsaAllocation();
+        uint64_t kernelStartPointer = kernel->getIsaOffsetInParentAllocation();
+//         if (this->heaplessModeEnabled) {
+            kernelStartPointer += isaAllocation->getGpuAddress();
+//         } else {
+//             kernelStartPointer += isaAllocation->getGpuAddressToPatch();
+//         }
+
+//         if (!kernel->requiresGenerationOfLocalIdsByRuntime()) {
+//             kernelStartPointer += kernel->getKernelDescriptor().entryPoints.skipPerThreadDataLoad;
+//         }
+
+        uint64_t iba = neoDevice->getMemoryManager()->getGfxPartition(neoDevice->getRootDeviceIndex())->getHeapBase(NEO::HeapIndex::heapInternalDeviceMemory);
+
+        LEVEL_ZERO_LAUNCH_KERNEL(iba, kernelStartPointer, kernelSize, kernelName);
+    }
+
     bool inOrderExecSignalRequired = false;
     bool inOrderNonWalkerSignalling = false;
 
diff --git a/level_zero/core/source/module/module_imp.cpp b/level_zero/core/source/module/module_imp.cpp
index e72cec9128..3e0ea86c05 100644
--- a/level_zero/core/source/module/module_imp.cpp
+++ b/level_zero/core/source/module/module_imp.cpp
@@ -41,6 +41,7 @@
 #include "shared/source/memory_manager/memory_operations_handler.h"
 #include "shared/source/memory_manager/unified_memory_manager.h"
 #include "shared/source/os_interface/os_context.h"
+#include "shared/source/os_interface/linux/probes.h"
 #include "shared/source/program/kernel_info.h"
 #include "shared/source/program/metadata_generation.h"
 #include "shared/source/program/program_initialization.h"
@@ -598,6 +599,19 @@ ze_result_t ModuleImp::initialize(const ze_module_desc_t *desc, NEO::Device *neo
     if (linkageSuccessful == false) {
         result = ZE_RESULT_ERROR_MODULE_LINK_FAILURE;
     }
+
+    if (linkageSuccessful && LEVEL_ZERO_COMPILE_KERNEL_ENABLED()) {
+        if (this->translationUnit->debugDataSize == 0) {
+            createDebugZebin();
+        }
+        auto blob = ArrayRef<const uint8_t>(reinterpret_cast<const uint8_t *>(this->translationUnit->debugData.get()),
+                                                                              this->translationUnit->debugDataSize);
+        auto dumpFilePath = dumpFileIncrement(reinterpret_cast<const char *>(blob.begin()),
+                                              blob.size(),
+                                              "/tmp/dumped_zebin_module", ".elf");
+        LEVEL_ZERO_COMPILE_KERNEL(dumpFilePath.c_str());
+    }
+
     return result;
 }
 
diff --git a/shared/source/CMakeLists.txt b/shared/source/CMakeLists.txt
index 70e2590d62..b854cca03a 100644
--- a/shared/source/CMakeLists.txt
+++ b/shared/source/CMakeLists.txt
@@ -79,6 +79,30 @@ endfunction()
 set(NEO_CORE_COMPILE_DEFS "")
 set(CORE_SOURCES ${CORE_SRCS_COREX_ALL_BASE})
 
+if(UNIX)
+  find_program(__DTRACE_FOUND dtrace)
+  if(__DTRACE_FOUND)
+
+    set(NEO_DTRACE_PROBES
+        ${CMAKE_CURRENT_SOURCE_DIR}/os_interface/linux/probes.d
+    )
+
+    message(STATUS "Using dtrace: dtrace")
+
+    set(NEO_DTRACE_OBJECTS "${NEO_DTRACE_PROBES}")
+    list(TRANSFORM NEO_DTRACE_OBJECTS REPLACE "[.]d$" ".o")
+
+    set(NEO_DTRACE_HEADERS "${NEO_DTRACE_PROBES}")
+    list(TRANSFORM NEO_DTRACE_HEADERS REPLACE "[.]d$" ".h")
+
+    add_custom_target(
+      DTRACE_PROBES ALL
+      COMMAND dtrace -G -s ${NEO_DTRACE_PROBES} -o ${NEO_DTRACE_OBJECTS}
+      COMMAND dtrace -h -s ${NEO_DTRACE_PROBES} -o ${NEO_DTRACE_HEADERS}
+    )
+  endif()
+endif()
+
 add_subdirectories()
 
 if(WIN32)
@@ -205,6 +229,13 @@ else()
   endif()
 endif()
 
+if(UNIX AND __DTRACE_FOUND)
+  target_link_libraries(
+    ${NEO_SHARED_RELEASE_LIB_NAME}
+    ${NEO_DTRACE_OBJECTS}
+  )
+endif()
+
 set_property(GLOBAL APPEND PROPERTY NEO_CORE_SRCS_LINK ${CORE_SRCS_LINK})
 
 get_property(NEO_SRCS_ENABLE_CORE GLOBAL PROPERTY NEO_SRCS_ENABLE_CORE)
diff --git a/shared/source/helpers/file_io.h b/shared/source/helpers/file_io.h
index f28d48d53f..b72f6fd680 100644
--- a/shared/source/helpers/file_io.h
+++ b/shared/source/helpers/file_io.h
@@ -26,7 +26,7 @@ size_t writeDataToFile(
 
 bool fileExists(const std::string &fileName);
 bool fileExistsHasSize(const std::string &fileName);
-void dumpFileIncrement(const char *data, size_t dataSize, const std::string &filename, const std::string &extension);
+std::string dumpFileIncrement(const char *data, size_t dataSize, const std::string &filename, const std::string &extension);
 
 #define USE_REAL_FILE_SYSTEM()                                                                                                                                                                                 \
     VariableBackup<decltype(NEO::IoFunctions::fopenPtr)> mockFopen(&NEO::IoFunctions::fopenPtr, [](const char *filename, const char *mode) -> FILE * {                                                         \
diff --git a/shared/source/helpers/file_io_load.cpp b/shared/source/helpers/file_io_load.cpp
index 852d2f1a48..49acb1161b 100644
--- a/shared/source/helpers/file_io_load.cpp
+++ b/shared/source/helpers/file_io_load.cpp
@@ -51,7 +51,7 @@ std::unique_ptr<char[]> loadDataFromFile(
     return ret;
 }
 
-void dumpFileIncrement(const char *data, size_t dataSize, const std::string &filename, const std::string &extension) {
+std::string dumpFileIncrement(const char *data, size_t dataSize, const std::string &filename, const std::string &extension) {
     auto filenameWithExt = filename + extension;
     int suffix = 0;
     while (fileExists(filenameWithExt)) {
@@ -59,4 +59,5 @@ void dumpFileIncrement(const char *data, size_t dataSize, const std::string &fil
         suffix++;
     }
     writeDataToFile(filenameWithExt.c_str(), std::string_view(data, dataSize));
+    return filenameWithExt;
 }
\ No newline at end of file
diff --git a/shared/source/os_interface/linux/probes.d b/shared/source/os_interface/linux/probes.d
new file mode 100644
index 0000000000..e4aa2ee996
--- /dev/null
+++ b/shared/source/os_interface/linux/probes.d
@@ -0,0 +1,4 @@
+provider level_zero {
+  probe launch_kernel(uint64_t iba, uint64_t baseAddress, uint64_t size, uintptr_t kernelName);
+  probe compile_kernel(uintptr_t pathName);
+}
